### ISSUE TITLE
`__class__` inside a lambda in class definition is reported as not defined

### ISSUE THREAD

**Issue by adamsol** [microsoft/pyright#11511]
**Describe the bug**
The code below is correct in Python (`A.f()` returns the class), but Pyright reports an error.

As an additional reference, there was a similar issue in Ruff: https://github.com/astral-sh/ruff/issues/20562

**Code or Screenshots**
```python
class A:
    f = lambda: __class__
```

`"__class__" is not defined  (reportUndefinedVariable)`

**VS Code extension or command-line**
VS Code, same in  [playground](https://pyright-play.net/?code=MYGwhgzhAECCBcBYAUNN0Bm0C81wFsAjAEzHmgH0LRIIqg)
